src/matrix.h

```diff
@@ -20,96 +21,465 @@
+		a.m[0][1] * b.m[3][0] + a.m[1][1] * b.m[3][1] + a.m[2][1] * b.m[3][2] + a.m[3][1] * b.m[3][3]
+	}, (const float[]) {
+		a.m[0][2] * b.m[0][0] + a.m[1][2] * b.m[0][1] + a.m[2][2] * b.m[0][2] + a.m[3][2] * b.m[0][3],
+		a.m[0][2] * b.m[1][0] + a.m[1][2] * b.m[1][1] + a.m[2][2] * b.m[1][2] + a.m[3][2] * b.m[1][3],
+		a.m[0][2] * b.m[2][0] + a.m[1][2] * b.m[2][1] + a.m[2][2] * b.m[2][2] + a.m[3][2] * b.m[2][3],
+		a.m[0][2] * b.m[3][0] + a.m[1][2] * b.m[3][1] + a.m[2][2] * b.m[3][2] + a.m[3][2] * b.m[3][3]
+	}, (const float[]) {
+		a.m[0][3] * b.m[0][0] + a.m[1][3] * b.m[0][1] + a.m[2][3] * b.m[0][2] + a.m[3][3] * b.m[0][3],
+		a.m[0][3] * b.m[1][0] + a.m[1][3] * b.m[1][1] + a.m[2][3] * b.m[1][2] + a.m[3][3] * b.m[1][3],
+		a.m[0][3] * b.m[2][0] + a.m[1][3] * b.m[2][1] + a.m[2][3] * b.m[2][2] + a.m[3][3] * b.m[2][3],
+		a.m[0][3] * b.m[3][0] + a.m[1][3] * b.m[3][1] + a.m[2][3] * b.m[3][2] + a.m[3][3] * b.m[3][3]
+	});
+}
+
+/**
+ * @return The linear interpolation of `a` and `b` using the specified fraction.
+ */
+static inline mat4_t __attribute__ ((warn_unused_result)) Mat4_Mix(const mat4_t a, const mat4_t b, float mix) {
+	mat4_t result;
+
+	for (size_t i = 0; i < lengthof(a.flat); i++) {
+		result.flat[i] = Mixf(a.flat[i], b.flat[i], mix);
+	}
+	
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wconditional-uninitialized"
+	return result;
+#pragma clang diagnostic pop
+}
+
+/**
+ * @return A perspective matrix with the specified parameters.
+ */
+static inline mat4_t __attribute__ ((warn_unused_result)) Mat4_FromFrustum(const float left, const float right, const float bottom, const float top, const float nearval, const float farval) {
+	const float rl = 1.f / (right - left);
+	const float tb = 1.f / (top - bottom);
+	const float nf = 1.f / (nearval - farval);
+	const float n2 = (nearval * 2.f);
+
+	return Mat4((const float[]) {
+		n2 * rl,				0.f,					0.f,						0.f,
+		0.f,					n2 * tb,				0.f,						0.f,
+		(right + left) * rl,	(top + bottom) * tb,	(farval + nearval) * nf,	-1.f,
+		0.f,					0.f,					(farval * n2) * nf,			0.f
+	});
+}
+
+/**
+ * @return An orthogonal matrix with the specified parameters.
+ */
+static inline mat4_t __attribute__ ((warn_unused_result)) Mat4_FromOrtho(const float left, const float right, const float bottom, const float top, const float nearval, const float farval) {
+	const float lr = 1.f / (left - right);
+	const float bt = 1.f / (bottom - top);
+	const float nf = 1.f / (nearval - farval);
+
+	return Mat4((const float[]) {
+		-2.f * lr,				0.f,					0.f,						0.f,
+		0.f,					-2.f * bt,				0.f,						0.f,
+		0.f,					0.f,					2.f * nf,					0.f,
+		(left + right) * lr,	(top + bottom) * bt,	(farval + nearval) * nf,	1.f
+	});
+}
+
+/**
+ * @return A matrix constructed from the specified translation, Euler angles and scale.
+ */
+static inline mat4_t __attribute__ ((warn_unused_result)) Mat4_FromOriginAnglesScale(const vec3_t origin, const vec3_t angles, const float scale) {
```

Comment:
Most of the functions use 'translate' instead of 'origin'. Mat4_FromTranslationRotationScale? That would be consistent with the functions below that do just one of those things.